apps/heureka-next/src/components/common/Filters/FilterSelect/FilterSelect.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useState } from "react"
+import { isEmpty } from "lodash"
+import { Button, InputGroup, SelectOption, Select, Stack } from "@cloudoperators/juno-ui-components"
+import { DropdownValue, Filter, SelectedFilter } from "../types"
+
+type FilterSelectProps = {
+  filters: Filter[]
+  onChange: (filter: SelectedFilter) => void
+}
+
+export const FilterSelect = ({ filters, onChange }: FilterSelectProps) => {
+  const [selectedFilterName, setSelectedFilterName] = useState<DropdownValue>("")
+  const [selectedFilterValue, setSelectedFilterValue] = useState<DropdownValue>("")
+  const filterValues = filters.find((filter) => filter.filterName === selectedFilterName)?.values
+
+  const handleValueChange = useCallback(
+    (value: DropdownValue) => {
+      if (!isEmpty(selectedFilterName) && !isEmpty(value)) {
+        onChange({
+          name: selectedFilterName as string, // we're sure that the value is a string
+          value: value as string, // we're sure that the value is a string
+        })
+        setSelectedFilterName("")
+        setSelectedFilterValue("")
+      }
+    },
+    [selectedFilterName, selectedFilterValue, onChange, setSelectedFilterName, setSelectedFilterValue]
```

Comment:
Just a question... Do we need the `setSelectedFilterName ` and  `setSelectedFilterValue ` since they are static as a dependency?

Reply:
I don't think so too. I've removed these. 
I'll have to fix `eslint` for this app after this PR as this could be caught by a lint rules at least as a warn during development.